compilers/concrete-compiler/compiler/lib/Dialect/FHELinalg/IR/FHELinalgOps.cpp

```diff
@@ -1592,6 +1592,67 @@ mlir::LogicalResult FancyAssignOp::verify() {
   return mlir::success();
 }
 
+mlir::LogicalResult BroadcastOp::verify() {
+  auto inputType =
+      this->getInput().getType().dyn_cast_or_null<mlir::RankedTensorType>();
+  auto outputType =
+      this->getOutput().getType().dyn_cast_or_null<mlir::RankedTensorType>();
+
+  auto inputElementType = inputType.getElementType();
```

Comment:
Is the cast via `dyn_cast_or_null` intentional? Using `dyn_cast_or_null` implies that the input and / or output types can be null, in which case the typed versions `inputType` / `outputType` would be null types. However, if that happens, `getElementType()` will fail.

Reply:
I'll update the code to use `dyn_cast`.